core/src/main/java/org/apache/iceberg/BaseMetadata.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.io.Serializable;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.iceberg.util.PropertyUtil;
+
+/** A base class for {@link TableMetadata} and {@link org.apache.iceberg.view.ViewMetadata} */
+public interface BaseMetadata extends Serializable {
```

Comment:
I'm not convinced that introducing this abstraction makes stuff easier. In most cases the methods in Hive just pass 1-2 things like schema / properties around

Reply:
@nastra we had discussed this a bit, with @pvary as well , you can take a look at the original pr  https://github.com/apache/iceberg/pull/8907/files  without the common metadata.  The main motivation is that HiveCatalog/ HiveTableOperations code is a bit hard to get right in terms of failure handling/locking, and cloning it is not that great.

Reply:
I understand that we want to reduce code duplication and it's a perfectly valid reason to think about having a base class, but https://github.com/apache/iceberg/pull/9852#discussion_r1516394903 is a bit concerning. What will be the behavior if e.g. `engine.hive.lock-enabled` or `engine.hive.enabled` deviates from its default value for a table but there's no option to control this on a view? Maybe I'm missing something obvious but it would be great to clarify how the final behavior in such cases will be

Reply:
OK good catch, its a problem on the other class (HiveOperationsBase).

@nk1506 can we remove this code please ?  For example, HiveOperationsBase defaut for hiveEngineEnabled() should be false , and only have HiveTableOperations should override this.

Reply:
It probably should not matter, as we will never store data in views, but it is definitely confusing.  cc @pvary to verify

Reply:
For engine.hive.lock-enabled, I think it is actually the desired behavior to do this on Hive and Views.  It will invoke a more optimized hive locking scheme, again @pvary being the author here.

Reply:
The `engine.hive.lock.enabed` should be available for views. The `engine.hive.enabled` is used for setting the storage handler class, which is irrelevant for views

Reply:
This has been taken care with `hiveEngineEnabled`
```
default boolean hiveEngineEnabled(Map<String, String> properties) {
    return false;
  }
  ```
`hiveLockEnabled` is using the same properties for table and view. 

```
default boolean hiveLockEnabled(Configuration conf, Map<String, String> properties) {
    if (properties.get(TableProperties.HIVE_LOCK_ENABLED) != null) {
      // We know that the property is set, so default value will not be used,
      return PropertyUtil.propertyAsBoolean(properties, TableProperties.HIVE_LOCK_ENABLED, false);
    }

    return conf.getBoolean(
        ConfigProperties.LOCK_HIVE_ENABLED, TableProperties.HIVE_LOCK_ENABLED_DEFAULT);
  }
```

Should I consider something different for `View`?